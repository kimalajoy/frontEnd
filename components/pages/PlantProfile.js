import React, {Component} from 'react';
import {
  View,
  Text,
  SafeAreaView,
  Image,
  TouchableOpacity,
  ImageBackground,
} from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import styles from '../../styles/styles';
import backgroundImg from '../../assets/plant_info_background.jpg';
import api from '../../api/plantAPI';


export default class PlantProfile extends Component {
  state = {
    plantInfo: {}
  }

  componentDidMount() {
    this.getPlantInfo(this.props.route.params.id);
  }

  getPlantInfo = (id) => {

    api.getGardenPlantById(id)
    .then(response => this.setState({plantInfo: response.data}))
    .catch(err => console.log(err))
  }

  render () {
    console.log('props', this.props)
    return (
      <SafeAreaView style={styles.container}>
        <ImageBackground source={backgroundImg} style={styles.greenCropBackground}>
          <View style={styles.plantInfoHeader}>
            <Text style={styles.plantName}>{this.state.plantInfo.plant_type}</Text>
            <Text style={styles.plantChildName}>{this.state.plantInfo.plant_name}</Text>
          </View>
          <View style={styles.transparentSubHeader}></View>
          <View style={styles.plantImgContainer}>
          {/* this image was removed and the icon was added - we can put the picture back once the image comes from API*/}
            <MaterialCommunityIcons
              style={styles.icon}
              name="flower"
              size={40} />
          </View>
          <View style={styles.plantContentContainer}>
            <View style={styles.plantChoresContent}>
              <Text style={styles.plantAttrLabel}>Last Watered On:</Text>
              <Text style={styles.plantAttrValue}>{this.state.plantInfo.last_watered}</Text>
              {/* placeholder below - will be adding this info from the API once it gets updated*/}
              <Text style={styles.plantAttrLabel}>Next Water in:</Text>
              <Text style={styles.plantAttrValue}>5 Days</Text>
              <Text style={styles.plantAttrLabel}>Harvest Date:</Text>
              <Text style={styles.plantAttrValue}>{this.state.plantInfo.harvest_date}</Text>
              {/* placeholder below - will be adding this info from the API once it gets updated*/}
              <Text style={styles.plantAttrLabel}>Harvest in:</Text>
              <Text style={styles.plantAttrValue}>200 Days</Text>
            </View>
          </View>
          <TouchableOpacity
            style={styles.backButtonContainer}
            onPress={() => {
              this.props.navigation.dangerouslyGetParent().setOptions({ tabBarVisible: true })
              this.props.navigation.navigate('MyGardenPage')
            }}>
            <Text style={styles.backButton}>Go Back to My Garden</Text>
          </TouchableOpacity>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}
