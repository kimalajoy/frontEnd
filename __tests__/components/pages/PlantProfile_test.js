import PlantProfile from '../../../components/pages/PlantProfile';
import renderer from 'react-test-renderer';
import React from 'react';


describe("PlantProfile Page", () => {
  let plantProfilePage;

  beforeAll(() => {
    plantProfilePage = renderer.create(<PlantProfile />)
  })

  it('PlantProfile Page does render', () => {
    expect(plantProfilePage).toMatchSnapshot();
  })
})
