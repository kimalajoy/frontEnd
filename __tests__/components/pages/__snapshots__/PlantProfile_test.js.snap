// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantProfile Page PlantProfile Page does render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fffefe",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/plant_info_background.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(93,162,91,.9)",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fafafa",
            "fontSize": 40,
          }
        }
      >
        Sunflower
      </Text>
      <Text
        style={
          Object {
            "color": "#fafafa",
            "fontSize": 20,
          }
        }
      >
        Jimothy
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, .5)",
          "flex": 2.5,
          "marginBottom": "-20%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": "-10%",
          "marginTop": "-5%",
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <Image
        style={
          Object {
            "borderColor": "orange",
            "borderRadius": 100,
            "borderWidth": 3,
            "height": 120,
            "width": 120,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 8,
          "height": "100%",
          "marginBottom": "10%",
          "marginTop": "-10%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(148, 126, 104, .9)",
            "borderRadius": 10,
            "height": "70%",
            "justifyContent": "space-around",
            "paddingLeft": "5%",
            "paddingRight": "2%",
            "paddingTop": "25%",
            "width": "85%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#E3CBB5",
              "fontSize": 24,
            }
          }
        >
          Last Watered On:
        </Text>
        <Text
          style={
            Object {
              "color": "#fafafa",
              "fontSize": 22,
              "marginBottom": "15%",
              "marginTop": "12%",
            }
          }
        >
          placeholder text
        </Text>
        <Text
          style={
            Object {
              "color": "#E3CBB5",
              "fontSize": 24,
            }
          }
        >
          Harvest Date:
        </Text>
        <Text
          style={
            Object {
              "color": "#fafafa",
              "fontSize": 22,
              "marginBottom": "15%",
              "marginTop": "12%",
            }
          }
        >
          placeholder text
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f8f8f8",
          "flex": 1.5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#383838",
            "fontSize": 30,
          }
        }
      >
        Go Back to My Garden
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
